### ISSUE TITLE
notif sent as root spawns extra dunst process

### ISSUE THREAD

**Issue by laur89** [dunst-project/dunst#510]
- Version: `1.3.0`
- Install type: `debian testing package (from official repo)`
- Distro and version: ` debian buster`

Sending messages via `notify-send` as regular user works perfectly, but as soon as it's done from a script as root like

```
sudo -u laur89 DISPLAY=:0 notify-send mymsg
su -l laur89 -c 'DISPLAY=:0 notify-send mymsg'
```

it spawns additional dunst daemon and displays the notification. This has 2 issues:

- additional daemon overhead;
- custom config is not picked up (ie dunst has default look-and-feel).

Note nothing was picked up by `bus-monitor path=/org/freedesktop/Notifications` when notify-send is used in abovementioned ways.

**Comment by tsipinakis** [comment]
I'm pretty sure this is an environment variable issue.

Dunst also needs to know where to find dbus, try setting the `DBUS_SESSION_BUS_ADDRESS` variable the same way you do with `DISPLAY`.

> custom config is not picked up (ie dunst has default look-and-feel).

I'm not sure about what's happening with this part thought. Dunst should be able to read the configuration file?

**Comment by laur89** [comment]
Cheers, exporting `DBUS_SESSION_BUS_ADDRESS` from the end of xinitrc does help with issue 2) (config bit).
Still spawns another process though.

**Comment by bebehei** [comment]
>  Still spawns another process though, and dbus-monitor is still not registering anything.

I have to admit this is really weird behavior and while reproducing your issue, I stumbled upon the same bit.

I assume in recent time, something has changed in the session handling. We should investigate this and [update the FAQs](https://dunst-project.org/faq/#sending-notifications-from-another-user-root).

**Comment by bebehei** [comment]
Ok, I found the solution.

On my ArchLinux machine, `notify-send` now only respects the `DBUS_SESSION_BUS_ADDRESS` env variable. This has to be set and match the current dunst process. Then every bogus behavior disappears.

Setting the `DISPLAY` variable has no effect anymore.

**Comment by bebehei** [comment]
> Setting the DISPLAY variable has no effect anymore.

Further research shows me, why I recommended the `DISPLAY` variable over the `DBUS_SESSION_BUS_ADDRESS`:

On older systems, the session bus was located under `/tmp/dbus-<10randomchars>`. I had to send notifications from the root user to the currently active user. This was almost impossible, to have reliably the actual bus there. For some reasons, the `DISPLAY` variable worked pretty well there and it had a big plus: It's actually static and usually won't change.

But, what I never actually have seen (I tested on Ubuntu 14.04 with Unity's native `notify-osd`) is the daemon overhead. Ubuntu does the same and also spawns another notify-osd daemon. But in comparison to `dunst` it's not important, as there is no real configuration of the OSD.

In the meantime, DBus session bus socket has changed to `/run/user/<UID>/bus` on recent distros (Ubu 18.04, too). This is a rather static path, so the initial problem with retrieving the abstract DBus socket path dissolves.

Sorry for giving wrong information in the FAQ. I'll fire a PR there.
